code/modules/reagents/reagents/addictions.dm

```diff
@@ -0,0 +1,52 @@
+#define TICK_ADDICTION_REMINDER 50
+
+/datum/addiction
+	var/name = "addiction"
+	var/tick //Tick goes up if you're not satiating your addiction, Tick goes down if you're satiating it
+	var/addiction = "memes" //Holds the reagent ID for whatever reagent you're addicted to, or reagent type if you want it to be covered by a subtype
+	var/mob/living/holder //The person with this addiction
+	var/addiction_multiplier = 1 //A way to simulate addictions getting 'worse' over time. As time goes by, this will increase, so you will need more of a chem. Can be used by addiction effects for greater effect
+	var/list/addiction_effects = list() //What effects this addiction will have on the holder. Organized as list(tick_to_trigger = effect)
+
+/datum/addiction/New(var/mob/living/new_holder, var/addicted)
+	..()
+	if(new_holder)
+		holder = new_holder
+		holder.addictions.Add(src)
+	if(addicted)
+		addiction = addicted
+
+/datum/addiction/Destroy()
+	if(holder)
+		holder.addictions.Remove(src)
+		holder = null
+	if(addiction_effects.len)
```

Comment:
the `.len` check is implicit in the for loop, it just doesn't run if len is 0, of course
however this will runtime if `addiction_effects` is already null
so you should just check for `addiction_effects`